sr_unix/util_output.c

```diff
@@ -347,6 +347,9 @@ caddr_t util_format(caddr_t message, va_list fao, caddr_t buff, ssize_t size, in
 					case 'Z': /* null teminated string */
 						GETFAOVALDEF(faocnt, fao, caddr_t, c, NULL);
 						length = c ? STRLEN(c) : 0;
+						break;
+					default:
+						assert(FALSE);
```

Comment:
I did not understand the commit comment for this module (**Add default clause to switch with an assert in it. Still doesn't prevent staticscan issue but next time through it can be marked ignorable**). Can you please explain what the original alert was about?

Reply:
As now noted in the commit message. This was an issue that surfaced in another routine (no idea which one) that one had to follow down through a few calls to figure out what the static scan was complaining about. This select was the area it was complaining about because it's choice for the switch value was none of the supported cases. I think it was the length variable a few lines down it ended up complaining about. I thought adding a default case didn't slow anything down and would allow us next time if this showed up to ignore it with better clarity. Oddly enough this issue showed up under the original higher level routine that made the call and not under util_output.c.

Reply:
I don't yet understand this explanation but I think I will once I see the newer commit message and code changes.

Reply:
Perhaps we should talk. There were no code changes and the commit line says pretty much the same as I posted earlier.

Reply:
As discussed, it is safer to not just add a default clause like you did but also set length to a huge value (MAXUINT4) in that case and then add an if (MAXUINT4 != length) check and execute all the following block of code only in that case. This way we address the issue cleanly. This is not a performance sensitive area of code so an extra if check is not a big thing.